cdap-hbase-compat-base/src/main/java/co/cask/cdap/replication/TableUpdater.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.replication;
+
+import co.cask.cdap.api.common.Bytes;
+import org.apache.hadoop.conf.Configuration;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Map;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Utility class for Replication Status Tool
+ */
+public abstract class TableUpdater {
+
+  private static final Logger LOG = LoggerFactory.getLogger(TableUpdater.class);
+  private ConcurrentHashMap<String, Long> cachedUpdates;
+  private Timer updateTimer;
+
+  public final byte[] rsID;
+  public final String columnQualifier;
+  public final Configuration conf;
+  public boolean tableExists = false;
+
+  public TableUpdater(String columnQualifier, final Configuration conf) {
+    this.columnQualifier = columnQualifier;
+    this.rsID = getUUIDArray(UUID.randomUUID());
+    this.conf = conf;
+
+    cachedUpdates = new ConcurrentHashMap<>();
+    setupTimer();
+  }
+
+  private byte[] getUUIDArray(UUID uuid) {
+    ByteBuffer byteBuffer = ByteBuffer.wrap(new byte[16]);
+    byteBuffer.putLong(uuid.getMostSignificantBits());
+    byteBuffer.putLong(uuid.getLeastSignificantBits());
+    return byteBuffer.array();
+  }
+
+  /**
+   * Synchronized method used by replication status tool TableUpdate class to add timestamps
+   * from coprocessors.
+   * @param regionName is the key for the Map
+   * @param time is the value for the Map
+   */
+  public synchronized void updateTime(String regionName, long time) {
+    if (!cachedUpdates.containsKey(regionName) || time > cachedUpdates.get(regionName)) {
+      cachedUpdates.put(regionName, time);
+    }
+  }
+
+  public final byte[] getRowKey(String regionName) {
+    ReplicationStatusKey replicationStatusKey = new ReplicationStatusKey(getRowPrefix(columnQualifier),
+                                                                         Bytes.toBytes(regionName), rsID);
+    return (replicationStatusKey.getKey());
+  }
+
+  public byte[] getRowPrefix(String columnQualifier) {
```

Comment:
instead of this method just store the pointerType as a protected byte[] and let the children access it directly.